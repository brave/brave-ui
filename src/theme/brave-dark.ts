import ITheme from './theme-interface'
import colors from './colors'
import defaultTheme from './brave-default'

const darkTheme: ITheme = {
  ...defaultTheme,
  name: 'Brave Dark',
  color: {
    ...defaultTheme.color,
    contextMenuBackground: colors.black,
    contextMenuForeground: colors.white,
    defaultControl: colors.white,
    defaultButtonBorder: colors.grey400
  }
}

export default darkTheme
