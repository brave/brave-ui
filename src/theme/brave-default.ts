import colors from './palette'
import ITheme from './theme-interface'

// Define colors, fonts, and sizes by purpose.
// Keys should not describe the value, but what they are to be used for.

const theme: ITheme = {
  color: {
    brandBrave: colors.orange400,
    brandBat: colors.blurple400,
    // brave controls
    brandBraveInteracting: colors.orange500,
    brandBraveActive: colors.orange200,
    brandBraveLight: colors.orange000,
    // bat controls
    brandBatInteracting: colors.blurple500,
    brandBatActive: colors.blurple200,
    // regular controls
    defaultControl: colors.grey800,
    defaultControlInteracting: colors.black,
    defaultControlActive: colors.grey700,
    // warning controls
    warn: colors.red500,
    warnInteracting: colors.red600,
    warnActive: colors.red700,
    // subtle controls
    subtle: colors.grey400,
    subtleInteracting: colors.grey500,
    subtleActive: colors.grey600,
    // disabled controls
    disabled: colors.grey300,
    // backgrounds
    primaryBackground: colors.white,
    secondaryBackground: colors.grey400
  },
  fontFamily: {
    heading: 'Poppins, sans-serif',
    body: 'Muli, sans-serif',
    system: '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif'
  }
}

export default theme
