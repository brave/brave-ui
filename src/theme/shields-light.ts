import ITheme from './theme-interface'
import defaultTheme from './brave-default'
import colors from './colors'

const shieldsLightTheme: ITheme = {
  ...defaultTheme,
  name: 'Shields Light',
  color: {
    ...defaultTheme.color,
    panelBackgroundSecondary: colors.neutral000
  }
}

export default shieldsLightTheme
