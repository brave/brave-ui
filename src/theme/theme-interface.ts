import { ThemedStyledProps } from 'styled-components'

export default interface IThemeProps {
  name: string,
  palette: { [key: string]: string }
  color: {
    contextMenuBackground: string
    contextMenuForeground: string
    brandBraveSubtleBackground: string
    brandBraveHover: string
    brandBraveMain: string
    brandBravePressed: string
    brandBraveOutline: string
    brandBrave: string
    brandBraveActive: string
    brandBatSubtleBackground: string
    brandBatHover: string
    brandBatMain: string
    brandBatPressed: string
    brandBatOutline: string
    brandBat: string
    brandBatInteracting: string
    brandBatActive: string
    detailDescription: string
    disabled: string
    disabledResourceBlocked: string
    primaryBackground: string
    secondaryBackground: string
    modalOverlayBackground: string
    defaultButtonText: string
    defaultButtonBorder: string
    defaultControl: string
    defaultControlInteracting: string
    defaultControlActive: string
    warn: string
    warnInteracting: string
    warnActive: string
    subtle: string
    subtleBackground: string
    subtleExclude: string
    subtleInteracting: string
    subtleActive: string
    text: string
    panelBackground: string
    panelBackgroundSecondary: string
    lionLogo: string
    inputBorder: string
    separatorLine: string
  }
  fontFamily: {
    heading: string
    body: string
    system: string
  }
}

export type BraveThemedStyledProps<T> = ThemedStyledProps<T, IThemeProps>
