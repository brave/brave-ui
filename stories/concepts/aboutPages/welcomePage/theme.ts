/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

const theme = {
  welcomePage: {
    fontFamily: '"Poppins", sans-serif',
    alignItems: 'center',
    justifyContent: 'center'
  },
  waveBackground: {
    height: '-webkit-fill-available',
    width: '-webkit-fill-available'
  },
  panel: {
    backgroundColor: 'rgba(255,255,255,0.95)',
    maxWidth: '600px',
    minHeight: '580px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    padding: '50px 60px'
  },
  braveLogo: {
    width: '90px'
  },
  paymentsImage: {
    width: '230px'
  },
  importImage: {
    width: '215px'
  },
  shieldsImage: {
    width: '185px'
  },
  featuresImage: {
    width: '300px'
  },
  title: {
    fontFamily: '"Poppins", sans-serif',
    fontSize: '32px',
    color: '#212121',
    margin: '30px 0 0',
    textAlign: 'center',
    lineHeight: '44px'
  },
  text: {
    fontFamily: '"Muli", sans-serif',
    fontSize: '18px',
    color: '#76777A',
    lineHeight: '34px',
    textAlign: 'center',
    margin: '20px 0'
  },
  mainButton: {
    textTransform: 'uppercase'
  },
  sideButton: {
    textTransform: 'uppercase',
    fontWeight: '500'
  },
  arrow: {
    display: 'inline-block',
    verticalAlign: 'middle'
  },
  content: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    flex: '1',
    marginBottom: '50px'
  },
  skip: {
    color: '#76777A',
    hoverColor: '#8B8A8E',
    textDecoration: 'underline',
    fontWeight: '300'
  },
  footer: {
    gridGap: '0',
    padding: '0'
  },
  footerColumnLeft: {
    alignItems: 'center',
    justifyContent: 'flex-start'
  },
  footerColumnCenter: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  footerColumnRight: {
    alignItems: 'center',
    justifyContent: 'flex-end'
  },
  bulletActive: {
    color: '#FB542B',
    hoverColor: '#FB542B',
    padding: '0 7px',
    fontSize: '40px'
  },
  bullet: {
    color: '#76777A',
    hoverColor: '#8B8A8E',
    padding: '0 7px',
    fontSize: '40px'
  }
} as any

export default theme
