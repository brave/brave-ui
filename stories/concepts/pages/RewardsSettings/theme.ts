/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

const theme = {
  page: {
    fontFamily: '"Poppins", sans-serif',
    color: '#fff',
    padding: '80px 60px 40px',
    background: 'linear-gradient(#4b3c6e, #000)'
  },
  trackersBlocked: {
    counterColor: '#f39030',
    descriptionColor: '#fff'
  },
  adsBlocked: {
    counterColor: '#fe521d',
    descriptionColor: '#fff'
  },
  httpsUpgrades: {
    counterColor: '#0796fa',
    descriptionColor: '#fff'
  },
  estimatedTime: {
    counterColor: '#999999',
    descriptionColor: '#fff'
  },
  clock: {
    color: 'rgba(255,255,255,0.8)'
  },
  clockContainer: {
    justifyContent: 'flex-end'
  },
  textualContainer: {
    maxWidth: '650px',
    margin: '60px auto 0',
    fontFamily: '"Muli", sans-serif'
  },
  media: {
    width: '80px',
    margin: '0 0 0 25px'
  },
  title: {
    fontFamily: '"Poppins", sans-serif',
    fontWeight: '400',
    color: '#fff'
  },
  italicText: {
    fontStyle: 'italic',
    fontSize: '15px'
  },
  text: {
    fontSize: '18px'
  },
  switchContainer: {
    margin: '40px 0 25px'
  }
}

export default theme
