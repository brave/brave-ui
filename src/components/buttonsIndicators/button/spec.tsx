/* global jest, expect, describe, it, afterEach */
import { shallow } from "enzyme";
import * as React from "react";
import { create } from "react-test-renderer";
import { TestThemeProvider } from "../../../theme";
import Button from "./index";

describe("Button tests", () => {
  const mockedOnClick = jest.fn();

  const baseComponent = (props?: object) => (
    <TestThemeProvider>
      <Button id="button" text="test" onClick={mockedOnClick} {...props} />
    </TestThemeProvider>
  );

  describe("basic tests", () => {
    it("matches the snapshot", () => {
      const component = baseComponent();
      const tree = create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("renders the component", () => {
      const wrapper = shallow(baseComponent());
      const assertion = wrapper.find("#button").length;
      expect(assertion).toBe(1);
    });
  });

  describe("behaviour tests", () => {
    it("calls onClick when clicked", () => {
      expect(mockedOnClick).not.toBeCalled();
      const wrapper = shallow(baseComponent());
      wrapper.find("#button").simulate("click");
      expect(mockedOnClick).toBeCalledTimes(1);
    });
  });
});
