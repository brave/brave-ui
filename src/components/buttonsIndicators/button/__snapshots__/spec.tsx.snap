// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests basic tests matches the snapshot 1`] = `
.c1 {
  --button-main-color: #242536;
  --button-main-color-hover: #131525;
  --button-main-color-active: #343546;
  --button-state-color: var(--button-main-color);
  --icon-size: 16px;
  --icon-spacing: 6px;
  --webkit-appearance: none;
  box-sizing: border-box;
  background: none;
  border: none;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Poppins,sans-serif;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 13px;
  border-radius: 20px;
  width: auto;
  min-width: 104px;
  padding: 11px 15px;
}

.c1:hover:enabled {
  --button-state-color: var(--button-main-color-hover);
}

.c1:active:enabled {
  --button-state-color: var(--button-main-color-active);
}

.c0 {
  color: #fff;
  background: var(--button-state-color);
  border: 1px solid var(--button-state-color);
}

.c2 {
  min-height: var(--icon-size);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-weight: 500;
  text-transform: none;
  line-height: 1;
}

<button
  className="c0 c1"
  id="button"
  size="medium"
  type="default"
>
  <div
    className="c2"
    id="button"
    size="medium"
    type="default"
  >
    test
  </div>
</button>
`;
